crates/agglayer-tries/src/roots.rs

```diff
@@ -0,0 +1,123 @@
+use std::fmt;
+
+use agglayer_primitives::{Digest, B256};
+use serde::{Deserialize, Serialize};
+
+macro_rules! define_root {
```

Comment:
Looking over the methods defined here, the macro does not contain anything specific to merkle tree roots. It could also be used to introduce strong type wrapper to other digest-typed values we have such as `CertificateId`.

```suggestion
macro_rules! define_digest_type {
```